src/Instances/AttributeChange.lua

```diff
@@ -0,0 +1,36 @@
+--!strict
+
+--[[
+	A special key for property tables, which allows users to connect to
+    an attribute change on an instance.
+]]
+
+local Package = script.Parent.Parent
+local PubTypes = require(Package.PubTypes)
+local logError = require(Package.Logging.logError)
+local xtypeof = require(Package.Utility.xtypeof)
+
+local function AttributeChanged(attributeName: string): PubTypes.SpecialKey
+	local attributeKey = {}
+	attributeKey.type = "SpecialKey"
+	attributeKey.kind = "AttributeChange"
+	attributeKey.stage = "observer"
+
+    function attributeKey:apply(callback: any, applyTo: Instance, cleanupTasks: {PubTypes.Task})
+        local ok, event = pcall(applyTo.GetAttributeChangedSignal, applyTo, attributeName)
+		if not ok then
+			logError("cannotConnectAttributeChange", nil, applyTo.ClassName, attributeName)
+		elseif typeof(callback) ~= "function" then
+			logError("invalidAttributeChangeHandler", nil, attributeName)
```

Comment:
The error priority here could be more elegant. Passing a non-function as callback is always a "simple" wrong usage at the user code definition, whereas failure to connect the attribute changed signal could have more deep rooted causes. Intuitively, I would argue for having the simplest errors checked first.

Reply:
I've moved the invalidHandler check outside of that scope so it should get checked first! Thanks :D